### PR TITLE
Introduce GH Actions for SPIRE CI

### LINKED ISSUE #2276: Move to GitHub Actions
Travis has begun charging the project lots of money to run CI/CD. I contacted the CNCF service desk to ask what we should do, and they indicated that we should move to GitHub Actions.

The project has been through a lot of CI/CD whiplash in the last couple years... hopefully we can finally put this to rest, as it sounds like CNCF will be able to provide us with a supported path that meets our needs.

### PR DESCRIPTION
**Pull request by evan2645**
This commit sets up GitHub Actions for SPIRE CI per CNCF request.

The nested rotation test is currently failing due to a problem with the docker workload attestor. The VMs that GitHub runs the Action jobs in ships with a docker that is set up to use custom cgroup names. This is causing the docker workload attestor to not recognize callers in docker containers. I think the fix will be to introduce a second agent config that has the correct matcher defined, and put a switch in the integration test script which detects GH build env and uses the alternate config.

GH sets many env variables by default, including one named `CI` that is always set to `true`. The full list is here: https://docs.github.com/en/actions/reference/environment-variables

I shipped a debug build to get the cgroup naming for the GH Actions docker installation. Here is a snippet from there:
`
12:hugetlb:/actions_job/8174f6a96af555d7e135d9bdbe4079da70fe5623cf8df4cddb40648519f826fc     11:perf_event:/actions_job/8174f6a96af555d7e135d9bdbe4079da70fe5623cf8df4cddb40648519f826fc     10:blkio:/actions_job/8174f6a96af555d7e135d9bdbe4079da70fe5623cf8df4cddb40648519f826fc     9:freezer:/actions_job/8174f6a96af555d7e135d9bdbe4079da70fe5623cf8df4cddb40648519f826fc     8:pids:/actions_job/8174f6a96af555d7e135d9bdbe4079da70fe5623cf8df4cddb40648519f826fc     7:cpu,cpuacct:/actions_job/8174f6a96af555d7e135d9bdbe4079da70fe5623cf8df4cddb40648519f826fc     6:net_cls,net_prio:/actions_job/8174f6a96af555d7e135d9bdbe4079da70fe5623cf8df4cddb40648519f826fc
5:rdma:/     
4:devices:/actions_job/8174f6a96af555d7e135d9bdbe4079da70fe5623cf8df4cddb40648519f826fc     3:cpuset:/actions_job/8174f6a96af555d7e135d9bdbe4079da70fe5623cf8df4cddb40648519f826fc     2:memory:/actions_job/8174f6a96af555d7e135d9bdbe4079da70fe5623cf8df4cddb40648519f826fc     1:name=systemd:/actions_job/8174f6a96af555d7e135d9bdbe4079da70fe5623cf8df4cddb40648519f826fc     
0::/system.slice/containerd.service
`

I am out on vacation starting tomorrow, so if someone is able to pick up this last piece, it would be much appreciated!

Fixes #2276

Signed-off-by: Evan Gilman <egilman@vmware.com>

### CODE REVIEW

**Comment by azdagron** [comment]
Awesome, @evan2645 ! I'm super excited for this.

I'll go ahead and push up a commit to fix the failing test.

**Comment by azdagron** [comment]
Confirmed that the commit fixes the integration tests on evan2645's fork.

**Comment by evan2645** [comment]
Thanks @azdagron! ❤️

File: .github/workflows/pr_build.yaml [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        uses: actions/cache@v2
+        with:
+          path: .build
+          key: ${{ runner.os }}-tools-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-tools-
+      - name: Build images
+        run: make images scratch-images
+      - name: Export images
+        run: docker save spire-server:latest-local spire-agent:latest-local k8s-workload-registrar:latest-local oidc-discovery-provider:latest-local spire-server-scratch:latest-local spire-agent-scratch:latest-local k8s-workload-registrar-scratch:latest-local | gzip > images.tar.gz
```
Comment (amartinezfayo) [inline .github/workflows/pr_build.yaml:130]: Is `oidc-discovery-provider-scratch` missing here?

Comment (azdagron) [reply .github/workflows/pr_build.yaml]: Good catch, @amartinezfayo! Updated.

**Review by amartinezfayo — APPROVED** [review]
Excited to see this merged!

**Pull request merged**